messaging/src/main/java/io/micronaut/messaging/MessagingApplication.java

```diff
@@ -80,13 +80,15 @@ public boolean isServer() {
     @NonNull
     public MessagingApplication start() {
         ApplicationContext applicationContext = getApplicationContext();
-        if (applicationContext != null && !applicationContext.isRunning()) {
-            try {
-                applicationContext.start();
-                applicationContext.publishEvent(new ApplicationStartupEvent(this));
-            } catch (Throwable e) {
-                throw new ApplicationStartupException("Error starting messaging server: " + e.getMessage(), e);
+        if (applicationContext != null) {
+            if (!applicationContext.isRunning()) {
+                try {
+                    applicationContext.start();
+                } catch (Throwable e) {
+                    throw new ApplicationStartupException("Error starting messaging server: " + e.getMessage(), e);
+                }
             }
+            applicationContext.publishEvent(new ApplicationStartupEvent(this));
```

Comment:
should this not be within he `if` block?

Reply:
No, it will not be triggered within the `isRunning` check. Netty has it the same but introduces custom events that extend application events.

Reply:
seems inconsistent with the `stop` implementation

Reply:
In what way? It looks fine to me.

Reply:
if the context is already running then calling `start()` will fire the message again even if the context is already running which is not the same as the previous behaviour?

Reply:
`ApplicationStartupEvent` is not fired when `start` is called. If I revert the code, the test will fail because the event is never published.